app/Platform/Actions/ComputeGameSearchTitlesAction.php

```diff
@@ -0,0 +1,387 @@
+            $variations[] = 'poc';
+        }
+
+        // Create simplified titles that don't have any special characters.
+        $simplifiedTitle = $this->simplifyTitle($title);
+        if ($simplifiedTitle !== $title) {
+            $variations[] = $simplifiedTitle;
+        }
+
+        // Handle titles starting with "The" by creating a version without it.
+        if (preg_match('/^The\s+(.+)$/i', $title, $matches)) {
+            $variations[] = $matches[1];
+            $variations[] = mb_strtolower($matches[1]);
+        }
+
+        return $variations;
+    }
+
+    /**
+     * Convert Roman numerals in titles to their numeric equivalents.
+     *
+     * This allows users to search for "Final Fantasy 7" and find "Final Fantasy VII".
+     * Only converts numerals that are word boundaries to avoid false positives.
+     */
+    private function convertRomanNumeralsToNumbers(string $title): string
+    {
+        // Need to reverse the order for regex to match longer numerals first (XX before X).
+        $romanNumerals = array_reverse(self::ROMAN_NUMERAL_MAP, true);
+
+        // Only replace Roman numerals that are complete words.
+        return preg_replace_callback(
+            '/\b(' . implode('|', array_keys($romanNumerals)) . ')\b/i',
+            function ($matches) {
+                $romanNumeral = strtoupper($matches[1]);
+
+                return self::ROMAN_NUMERAL_MAP[$romanNumeral] ?? $romanNumeral;
+            },
+            $title
+        );
+    }
+
+    /**
+     * Convert numbers in titles to their Roman numeral equivalents.
+     *
+     * This allows users to search for "Final Fantasy VII" and find "Final Fantasy 7".
+     * Limited to numbers 1-20 which covers most common game series.
+     */
+    private function convertNumbersToRomanNumerals(string $title): string
+    {
+        // Flip the map to go from numbers to Roman numerals (eg: '7' -> 'VII').
+        $numberToRomanMap = array_flip(self::ROMAN_NUMERAL_MAP);
+
+        return preg_replace_callback(
+            '/\b(' . implode('|', array_keys($numberToRomanMap)) . ')\b/',
+            function ($matches) use ($numberToRomanMap) {
+                return $numberToRomanMap[$matches[1]] ?? $matches[1];
+            },
+            $title
+        );
+    }
+
+    /**
+     * Generate common abbreviations for game titles.
+     *
+     * Creates shortened versions like "FF7" for "Final Fantasy VII" or "SM64" for
+     * "Super Mario 64". This handles the common way users search for games using
+     * series abbreviations combined with numbers.
+     */
+    private function generateAbbreviations(string $title): array
+    {
+        $abbreviations = [];
+
+        // Map of series names to their common abbreviations.
+        // It doesn't matter if some of these overlap, and these aren't case-sensitive.
+        // This is just a way we can influence how heavily Meilisearch weighs certain terms.
```

Comment:
It almost seems like this should be a field on the GameData record where an admin can add specific aliases for a game.

Additionally, that would provide a way to add new entries without having to modify the code.

The first one I could come up with is `RotK` (Romance of the Three Kingdoms).

While many of these are common, I don't think I've ever seen anyone abbreviate F-Zero or Starfox.